src/steamship/agents/context/chathistory.py

```diff
@@ -0,0 +1,225 @@
+                        text=initial_system_prompt,
+                        tags=[Tag(kind=TagKind.ROLE, name=RoleTag.SYSTEM)],
+                    )
+                )
+
+            file = File.create(
+                client=client,
+                blocks=blocks,
+                tags=tags,
+            )
+        return ChatHistory(file)
+
+    def append_user_message(
+        self,
+        text: str = None,
+        tags: List[Tag] = None,
+        content: Union[str, bytes] = None,
+        url: Optional[str] = None,
+        mime_type: Optional[MimeTypes] = None,
+    ) -> Block:
+        """Append a new block to this with content provided by the end-user."""
+        tags = tags or []
+        tags.append(
+            Tag(
+                kind=TagKind.CHAT, name=ChatTag.ROLE, value={TagValueKey.STRING_VALUE: RoleTag.USER}
+            )
+        )
+        return self.file.append_block(
+            text=text, tags=tags, content=content, url=url, mime_type=mime_type
+        )
+
+    def append_system_message(
+        self,
+        text: str = None,
+        tags: List[Tag] = None,
+        content: Union[str, bytes] = None,
+        url: Optional[str] = None,
+        mime_type: Optional[MimeTypes] = None,
+    ) -> Block:
+        """Append a new block to this with content provided by the system, i.e., instructions to the assistant."""
+        tags = tags or []
+        tags.append(
+            Tag(
+                kind=TagKind.CHAT,
+                name=ChatTag.ROLE,
+                value={TagValueKey.STRING_VALUE: RoleTag.SYSTEM},
+            )
+        )
+        return self.file.append_block(
+            text=text, tags=tags, content=content, url=url, mime_type=mime_type
+        )
+
+    def append_agent_message(
+        self,
+        text: str = None,
+        tags: List[Tag] = None,
+        content: Union[str, bytes] = None,
+        url: Optional[str] = None,
+        mime_type: Optional[MimeTypes] = None,
+    ) -> Block:
+        """Append a new block to this with content provided by the agent, i.e., results from the assistant."""
+        tags = tags or []
+        tags.append(
+            Tag(
+                kind=TagKind.CHAT,
+                name=ChatTag.ROLE,
+                value={TagValueKey.STRING_VALUE: RoleTag.ASSISTANT},
+            )
+        )
+        return self.file.append_block(
+            text=text, tags=tags, content=content, url=url, mime_type=mime_type
+        )
+
+    @property
+    def last_user_message(self) -> Optional[Block]:
+        for block in self.file.blocks[::-1]:
+            if block.chat_role == RoleTag.USER:
+                return block
+        return None
+
+    @property
+    def last_system_message(self) -> Optional[Block]:
+        for block in self.file.blocks[::-1]:
+            if block.chat_role == RoleTag.SYSTEM:
+                return block
+        return None
+
+    @property
+    def last_agent_message(self) -> Optional[Block]:
```

Comment:
These helpers feel super useful. 

I think for tool -> tool -> tool pipelines, we might want one that's `last_message` which loads any of a set of roles? E.g. the tool's input might be the last tool's output OR last user input, whichever is later?